HIF.md

```diff
@@ -0,0 +1,823 @@
+    will use the `timestamp_us` value from [`CNF_DATA_TX`][tx-cnf] and
+    [`IND_DATA_RX`][rx].
+ - `uint24_t ufsi` (from UTT-IE)
+ - `uint8_t dwell_interval` (from US-IE)
+
+Only present if `FHSS_TYPE_LFN_UC`:
+
+ - `uint64_t lutt_timestamp_us`  
+    Timestamp associated with the last received LUTT-IE from this node. The host
+    will use the `timestamp_us` value from [`CNF_DATA_TX`][tx-cnf] and
+    [`IND_DATA_RX`][rx].
+ - `uint16_t slot` (from LUTT-IE)
+ - `uint24_t interval_offset_ms` (from LUTT-IE)
+ - `uint24_t interval_ms` (from LUS-IE)
+
+Only present if `FHSS_TYPE_LFN_PA`:
+
+ - `uint64_t lnd_timestamp_us`  
+    Timestamp associated with the last received LND-IE from this node. The host
+    will use the `timestamp_us` value from [`IND_DATA_RX`][rx].
+ - `uint24_t response_delay_ms` (from LND-IE)
+ - `uint8_t  slot_duration_ms` (from LND-IE)
+ - `uint8_t  slot_count` (from LND-IE)
+ - `uint16_t slot_first` (from LND-IE)
+
+Other FHSS types have no additional fields.
+
+Only present if `FHSS_DEFAULT == 0`:
+
+ - `struct chan_seq`  
+   See ["Channel Sequence"][chan-seq].
+
+For each bit set in `FRAME_COUNTERS`:
+
+ - `uint32_t frame_counter`  
+   Minimum frame counter to accept when verifying acknowlegment frames. The
+   associated key index maps to the bit offset (from 1 to 7). See
+   ["Security"][sec].
+
+Only present if `MODE_SWITCH`:
+
+ - `struct rate_config[4]`  
+    Fixed length and ordered array of rates to attempt until the transmission
+    is successful, or all rates have been tried.
+     - `uint8_t phy_mode_id`: Wi-SUN _PhyModeId_, must be in the group selected
+        with [`SET_RADIO`][rf-set].
+     - `uint8_t tx_attempts`: Maximum number of attempts for this rate, try the
+        next entry once this is exceeded. Note that this overrides
+        `macMaxFrameRetries` described in ["Channel Access and Retries"][cca].
+     - `int8_t tx_power_dbm`: TX power to use with this entry, saturates to the
+        value configured with [`SET_RADIO_TX_POWER`][rf-pow].
+
+### `0x12 CNF_DATA_TX`
+
+Status of an earlier data request ([`REQ_DATA_TX`][tx-req]). Also returns the
+IEEE 802.15.4 acknowledgement frame if the request had the acknowledgement
+request bit set in the header.
+
+ - `uint8_t handle`  
+    Frame handle, as provided in [`REQ_DATA_TX`][tx-req].
+
+ - `uint8_t status`  
+    See the table below for a list of status codes. The fields below may be
+    invalid if `status != 0`.
+
+ - `uint16_t frame_len`  
+    Length of the IEEE 802.15.4 acknowledgment frame if received.
+
+ - `uint8_t frame[]`  
+    Decrypted IEEE 802.15.4 acknowledgment frame (see ["Security"][sec]).
+    <!-- TODO: Explain EDFE. -->
+
+ - `uint64_t timestamp_us`  
+    The timestamp (relative to the date of the RCP reset) when the frame has
+    been received.
+
+ - `uint8_t lqi`  
+    Received Link Quality Indicator (LQI) as reported by [RAIL][lqi] (ACK
+    received only).
```

Comment:
The API is already in use, it cannot be simply dropped unfortunately.

Reply:
Because it is not used and may be misleading. It's opening the door to questions like "why my parent is not that router instead, its lqi is better than the one chosen by RPL"

Reply:
This API has been designed to allow customers to do customisation. This is the reason why we expose all the data we can rather than only the needed one.